content/en/security_platform/application_security/troubleshooting.md

```diff
@@ -0,0 +1,508 @@
+```
+
+**Note:** The `dd-test-scanner-log` value is supported in the most recent releases and in the application security event configuration file version >= 1.2.5.
+
+{{< /programming-lang >}}
+{{< programming-lang lang="go" >}}
+
+ ```bash
+ for ((i=1;i<=200;i++)); 
+do
+# Target existing service’s routes
+curl https://your-application-url/existing-route -A Arachni/v1.0;
+# Target non existing service’s routes
+curl https://your-application-url/non-existing-route -A Arachni/v1.0;
+done
+```
+
+{{< /programming-lang >}}
+{{< programming-lang lang="ruby" >}}
+
+ ```bash
+ for ((i=1;i<=200;i++)); 
+do
+# Target existing service’s routes
+curl https://your-application-url/existing-route -A Arachni/v1.0;
+# Target non existing service’s routes
+curl https://your-application-url/non-existing-route -A Arachni/v1.0;
+done
+```
+
+{{< /programming-lang >}}
+{{< programming-lang lang="PHP" >}}
+
+```bash
+for ((i=1;i<=200;i++)); 
+do
+# Target existing service’s routes
+curl https://your-application-url/existing-route -A dd-test-scanner-log;
+# Target non existing service’s routes
+curl https://your-application-url/non-existing-route -A dd-test-scanner-log;
+done
+```
+
+**Note:** The `dd-test-scanner-log` value is supported in the most recent releases and in the application security event configuration file version >= 1.2.5.
+
+{{< /programming-lang >}}
+{{< programming-lang lang="NodeJS" >}}
+
+```bash
+for ((i=1;i<=200;i++)); 
+do
+# Target existing service’s routes
+curl https://your-application-url/existing-route -A dd-test-scanner-log;
+# Target non existing service’s routes
+curl https://your-application-url/non-existing-route -A dd-test-scanner-log;
+done
+```
+**Note:** The `dd-test-scanner-log` value is supported in the most recent releases and in the application security event configuration file version >= 1.2.5.
+
+{{< /programming-lang >}}
+{{< /programming-lang-wrapper >}}
+
+A few minutes after you enable your application and exercise it, and if it's successful, threat information appears in the [Trace and Signals Explorer][2].
+
+{{< img src="/security_platform/application_security/application-security-signal.png" alt="Security Signal details page showing tags, metrics, suggested next steps, and attacker IP addresses associated with a threat." style="width:100%;" >}}
+
+### Check if required tracer integrations are deactivated
+
+Application Security relies on certain tracer integrations. If they are deactivated, Application Security won't work. To see if there are deactivated integrations, look for `disabled_integrations` in your [startup logs][8].
+
+The required integrations vary by language.
+
+{{< programming-lang-wrapper langs="java,.NET,go,ruby,PHP,NodeJS" >}}
+{{< programming-lang lang="java" >}}
+
+For [Java][1], the required integrations are the following:
+
+- servlet
+- servlet-2
+- servlet-3
+- servlet-request-body
+- grizzly
+- grizzly-char-body
+- grizzly-byte-body
+- grizzly-filterchain
+- spring-web
+- spring-webflux
```

Comment:
This list needs to be updated. `jersey`, `jetty`, `ratpack-request-body`, `tomcat`, `resteasy`. It's a bit inconsistent, I now realize, but the instrumenters with `grizzly-char-body` and `grizzly-byte-body` were renamed also to simply `grizzly`.

Reply:
Hey @cataphract I updated the list and also the wording as we discussed over DM.